### PR TITLE
add db support for project component phases

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/11427

## Testing
**URL to test:** 

local

**Steps to test:**
- restart hasura cluster and open console
- go to `moped_proj_components` table
- edit an existing record and add a `phase_id` and `subphase_id` (e.g., '1') and populate the `phase_end` and `date_added` fields
- nested data for the associated phase and subphase should be accessible under the `moped_phase` and `moped_subphase` column

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

**Review by johnclary — APPROVED** [review]
i tested this via the hasura console - this looks great! 

🚢✨

File: moped-database/migrations/1675978843251_alter_table_public_moped_proj_components_add_column_phase_id/up.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ null;
+alter table "public"."moped_proj_components" add column "phase_end" timestamptz
+ null;
+alter table "public"."moped_proj_components" add column "date_added" timestamptz
+ not null default now();
+alter table "public"."moped_proj_components"
+  add constraint "moped_proj_components_phase_id_fkey"
+  foreign key ("phase_id")
+  references "public"."moped_phases"
+  ("phase_id") on update cascade on delete set null;
```
Comment (frankhereford) [inline moped-database/migrations/1675978843251_alter_table_public_moped_proj_components_add_column_phase_id/up.sql:13]: 👍

File: moped-database/migrations/1675978843251_alter_table_public_moped_proj_components_add_column_phase_id/down.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-- drop all new columns (this will also drop constraints)
+alter table "public"."moped_proj_components" drop column "phase_id";
+alter table "public"."moped_proj_components" drop column "subphase_id";
+alter table "public"."moped_proj_components" drop column "phase_end";
+alter table "public"."moped_proj_components" drop column "date_added";
```
Comment (frankhereford) [inline moped-database/migrations/1675978843251_alter_table_public_moped_proj_components_add_column_phase_id/down.sql:5]: The least important thing in the world, but if you do end up editing these files again, you may want to drop a new line at the end of this file.

**Review by frankhereford — APPROVED** [review]
This works great, thank you Tilly! I tested this via a connection directly to the DB with the following:

```
select * from moped_phases where phase_id = 1;
select * from moped_subphases where subphase_id = 1;
select * from moped_proj_components where project_component_id = 2;
update moped_proj_components set phase_id = 1, subphase_id = 1  where project_component_id = 2;
select * from moped_proj_components where project_component_id = 2;
```

And then via a graphql query via the console:

```
query MyQuery {
  moped_proj_components(where: {project_component_id: {_eq: 2}}) {
    description
    interim_project_component_id
    phase_end
    phase_id
    subphase_id
    moped_subphase {
      subphase_id
      subphase_description
    }
    moped_phase {
      phase_id
      phase_description
    }
  }
}
```

🚢🚢🚢

File: moped-database/migrations/1675978843251_alter_table_public_moped_proj_components_add_column_phase_id/down.sql (hunk 2 above)
Comment (tillyw) [reply moped-database/migrations/1675978843251_alter_table_public_moped_proj_components_add_column_phase_id/down.sql]: thanks @frankhereford ! added the newline

**Review by mddilley — APPROVED** [review]
Followed the steps, and this works splendidly 🙌 ✨ I also tested with a query and could get any phase or subphase info that I wanted about a component. 🚀

**Pull request merged**